src/agent/block_store_services/block_store_fs.js

```diff
@@ -230,22 +254,39 @@ class BlockStoreFs extends BlockStoreBase {
                 usage.count -= 1;
 
                 // also make sure we do not leave old .meta files on overwrite
-                const meta_path = this._get_block_meta_path(block_md.id);
                 const overwrite_meta_stat = await nb_native().fs.stat(fs_context, meta_path).catch(ignore_not_found);
                 if (overwrite_meta_stat) {
                     usage.size -= overwrite_meta_stat.size;
-                    await nb_native().fs.unlink(fs_context, meta_path).catch(ignore_not_found);
+                    if (!this.xattr_fallback) {
+                        // Remove the file iff we are not in
+                        await nb_native().fs.unlink(fs_context, meta_path).catch(ignore_not_found);
+                    }
                 }
             }
         }
 
         try {
-            await nb_native().fs.writeFile(fs_context, block_path, data, {
-                xattr,
-                xattr_try,
-                xattr_need_fsync,
-            });
+            if (this.xattr_fallback) {
+                await Promise.all([
+                    nb_native().fs.writeFile(fs_context, block_path, data),
+                    nb_native().fs.writeFile(fs_context, meta_path, Buffer.from(block_md_data, 'utf8')),
```

Comment:
I think we need to fix the usage.size for this case. see the code from before the change -
https://github.com/noobaa/noobaa-core/blob/e726c111d3389e8afaa350a8d4a4d79c85fa0bec/src/agent/block_store_services/block_store_fs.js#L150